File: unidbg-android/src/main/java/com/vortexdbg/linux/android/dvm/wrapper/DvmLong.java
package com.vortexdbg.linux.android.dvm.wrapper;

import com.vortexdbg.linux.android.dvm.DvmObject;
import com.vortexdbg.linux.android.dvm.VM;

public class DvmLong extends DvmObject<Long> {

    public static DvmLong valueOf(VM vm, long i) {
        return new DvmLong(vm, i);
    }

    private DvmLong(VM vm, Long value) {
        super(vm.resolveClass("java/lang/Long"), value);
    }
}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/android/dvm/wrapper/DvmLong.kt
package com.vortexdbg.linux.android.dvm.wrapper

import com.vortexdbg.linux.android.dvm.DvmObject
import com.vortexdbg.linux.android.dvm.VM

class DvmLong private constructor(vm: VM, value: Long) : DvmObject<Long>(vm.resolveClass("java/lang/Long"), value) {

    companion object {
        @JvmStatic
        fun valueOf(vm: VM, i: Long): DvmLong {
            return DvmLong(vm, i)
        }
    }
}
